hkmc2/shared/src/test/mlscript/objbuf/Basics.mls

```diff
@@ -0,0 +1,223 @@
+:js
+
+import "../../mlscript-compile/ObjectBuffer.mls"
+open ObjectBuffer
+open annotations
+
+
+:sjs
+@buffered
+class A(x) with
+  fun f(y) = x + y
+//│ JS (unsanitized):
+//│ let A1;
+//│ globalThis.Object.freeze(class A {
+//│   static {
+//│     A1 = this
+//│   }
+//│   constructor() {}
+//│   static {
+//│     this.size = 1;
+//│   }
+//│   static ctor(buf, idx) {
+//│     return (x) => {
+//│       let idx1;
+//│       idx1 = idx + 0;
+//│       buf.buf[idx1] = x;
+//│       return idx
+//│     }
+//│   } 
+//│   static f(buf, idx) {
+//│     return (y) => {
+//│       let idx1;
+//│       idx1 = idx + 0;
+//│       return buf.buf.at(idx1) + y
+//│     }
+//│   }
+//│   #x;
+//│   toString() { return runtime.render(this); }
+//│   static [definitionMetadata] = ["class", "A"]; 
+//│ });
+
+:sjs
+@bufferable
+class A2(x)
+//│ JS (unsanitized):
+//│ let A21;
+//│ A21 = function A2(x) {
+//│   return globalThis.Object.freeze(new A2.class(x));
+//│ };
+//│ globalThis.Object.freeze(class A2 {
+//│   static {
+//│     A21.class = this
+//│   }
+//│   constructor(x) {
+//│     this.#x = x;
+//│   }
+//│   static {
+//│     this.size = 1;
+//│   }
+//│   static ctor(buf, idx) {
+//│     return (x) => {
+//│       let idx1;
+//│       idx1 = idx + 0;
+//│       buf.buf[idx1] = x;
+//│       return idx
+//│     }
+//│   }
+//│   #x;
+//│   toString() { return runtime.render(this); }
+//│   static [definitionMetadata] = ["class", "A2", [null]]; 
+//│ });
+
+:ge
+@buffered
+@buffered
+class B1(x)
+//│ ╔══[COMPILATION ERROR] Only one of bufferable annotation is allowed.
+//│ ║  l.74: 	@buffered
+//│ ║        	 ^^^^^^^^
+//│ ║  l.75: 	@buffered
+//│ ╙──      	^^^^^^^^^
+
+:ge
+@buffered
+@bufferable
+class B2(x)
+//│ ╔══[COMPILATION ERROR] Only one of bufferable annotation is allowed.
+//│ ║  l.84: 	@buffered
+//│ ║        	 ^^^^^^^^
+//│ ║  l.85: 	@bufferable
+//│ ╙──      	^^^^^^^^^^^
+
+:w
+@bufferable
+module B3
+//│ ╔══[WARNING] This annotation has no effect.
+//│ ║  l.94: 	@bufferable
+//│ ╙──      	 ^^^^^^^^^^
+
+:w
+@bufferable
+object B3
+//│ ╔══[WARNING] This annotation has no effect.
+//│ ║  l.101: 	@bufferable
+//│ ╙──       	 ^^^^^^^^^^
+
+
+let buf = new DefaultObjectBuffer(10)
+//│ buf = DefaultObjectBuffer(_)
+
+let obj1 = buf.mkNew(A)(3)
+//│ obj1 = 0
+
+let obj2 = buf.mkNew(A)(5)
+//│ obj2 = 2
+
+A.f(buf, obj1)(1)
+//│ = 4
+
+A.f(buf, obj2)(1)
+//│ = 6
+
+buf.del(A, obj1)
+
+let obj3 = buf.mkNew(A)(10)
+//│ obj3 = 0
+
+// UAF
+A.f(buf, obj1)(1)
+//│ = 11
+
+A.f(buf, obj3)(1)
+//│ = 11
+
+
+:sjs
```

Comment:
Does this show anything interesting that's not shown above? Otherwise, let's remove it.